modules/sdk/gradle-plugins-defaults/src/main/java/com/liferay/gradle/plugins/defaults/internal/LiferayBaseDefaultsPlugin.java

```diff
@@ -17,23 +17,77 @@
 import com.liferay.gradle.plugins.BaseDefaultsPlugin;
 import com.liferay.gradle.plugins.LiferayBasePlugin;
 import com.liferay.gradle.plugins.defaults.internal.util.GradleUtil;
+import com.liferay.gradle.plugins.defaults.tasks.CleanBuildProfileTask;
+import com.liferay.gradle.plugins.defaults.tasks.SetBuildProfileTask;
 import com.liferay.gradle.plugins.extensions.LiferayExtension;
 
 import java.io.File;
 
 import org.gradle.api.Action;
 import org.gradle.api.Plugin;
 import org.gradle.api.Project;
+import org.gradle.api.Task;
+import org.gradle.api.UnknownTaskException;
+import org.gradle.api.logging.Logger;
+import org.gradle.api.tasks.TaskContainer;
 
 /**
  * @author Andrea Di Giorgi
+ * @author Christopher Bryan Boyd
+ * @author Gregory Amerson
  */
 public class LiferayBaseDefaultsPlugin
 	extends BaseDefaultsPlugin<LiferayBasePlugin> {
 
 	public static final Plugin<Project> INSTANCE =
 		new LiferayBaseDefaultsPlugin();
 
+	@Override
+	public void apply(Project project) {
+		super.apply(project);
+
+		Logger logger = project.getLogger();
+
+		Project parentProject = project.getParent();
+
+		TaskContainer taskContainer = parentProject.getTasks();
+
+		if (taskContainer.findByPath(
+				parentProject.getPath() + ":" +
+					CleanBuildProfileTask.CLEAN_BUILD_PROFILE_TASK_NAME) ==
+						null) {
+
+			taskContainer.create(
```

Comment:
Why are you checking if the parent does not have the task and then adding the task to the child, but then later on in line 74 you are adding the task regardless?